src/controller/CHIPDevice.cpp

```diff
@@ -398,6 +398,22 @@ void Device::CancelResponseHandler(uint8_t seqNum)
     mCallbacksMgr.CancelResponseCallback(mDeviceId, seqNum);
 }
 
+void Device::AddIMResponseHandler(Callback::Cancelable * onSuccessCallback, Callback::Cancelable * onFailureCallback)
+{
+    // This is a bit tricky, we try to assume that chip::NodeId is uint64_t so the pointer can be used as a NodeId for CallbackMgr.
+    static_assert(std::is_same<chip::NodeId, uint64_t>::value, "chip::NodeId is not uint64_t");
+    mCallbacksMgr.AddResponseCallback(reinterpret_cast<chip::NodeId>(static_cast<app::Command *>(mCommandSender)),
+                                      0 /* seqNum, always 0 for IM */, onSuccessCallback, onFailureCallback);
```

Comment:
`0` is not going to work. If someone registers an other handler that the first one will be cancelled. Then when it will be retrieved into `CHIPClientCallbacks` it may crash.

Reply:
The IM Delegate is using the address of mCommandSender object instead of nodeid + seqnum in ember, thus the handler is always bind to the command sender instead of node id, in this case, this callback will always retreve the uptodate callback.

Reply:
Sorry, maybe I'm missing something obvious :( 

Assuming a client send 2 commands to a particular `CHIPDevice`, how would you make the difference between the 2 responses using the address of  `mCommandSender` ?

Reply:
Yes, you are right, we should use the CommandSenderObj and the sequence number to identify different status in one invoke action per spec.
However, per current controller implementation, we just encode and send one command at a time (in CHIPClusters.cpp), so just use command sender to identify responses is ok for current tests cases, but some refactors is required.

Reply:
Well we already have bug reports because humans are sending a bunch of commands (for example https://github.com/project-chip/connectedhomeip/issues/5678 which is the same command over and over, but I would not be surprised that some send different commands)  without waiting for the result.  While the current approach will likely works for passing the test it will probably not work in real conditions ?

Can't we use the exchange id or something like that ?